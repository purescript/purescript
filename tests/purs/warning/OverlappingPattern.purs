-- @shouldWarnWith OverlappingPattern
-- @shouldWarnWith OverlappingPattern
module Main where

data X = A | B

pat1 :: X -> Boolean
pat1 A = true
pat1 A = true
pat1 B = false

pat2 :: X -> Boolean
pat2 A = true
pat2 _ = false
pat2 B = false
