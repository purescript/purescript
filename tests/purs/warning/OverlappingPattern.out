Warning 1 of 2:

  in module [33mMain[0m
  at tests/purs/warning/OverlappingPattern.purs:12:1 - 12:21 (line 12, column 1 - line 12, column 21)

    A case expression contains unreachable cases:

    B

  in value declaration [33mpat2[0m

  See https://github.com/purescript/documentation/blob/master/errors/OverlappingPattern.md for more information,
  or to contribute content related to this warning.

Warning 2 of 2:

  in module [33mMain[0m
  at tests/purs/warning/OverlappingPattern.purs:7:1 - 7:21 (line 7, column 1 - line 7, column 21)

    A case expression contains unreachable cases:

    A

  in value declaration [33mpat1[0m

  See https://github.com/purescript/documentation/blob/master/errors/OverlappingPattern.md for more information,
  or to contribute content related to this warning.

